flink-elastic-catalog/src/test/java/com/getindata/flink/connector/jdbc/catalog/ElasticCatalogTest.java

```diff
@@ -577,21 +577,137 @@ public void testGetTablePartitionBySpecialCharacterColumn() throws TableNotExist
     public void testPreservesColumnsOrder() throws TableNotExistException {
         // given
         String url = String.format("jdbc:elasticsearch://%s:%d", container.getHost(),
-            container.getElasticPort());
+                container.getElasticPort());
 
         // when
         ElasticCatalog catalog = new ElasticCatalog("test-catalog", "test-database", USERNAME, PASSWORD, url);
 
         // then
         String[] actualColumnNames = catalog.getTable(new ObjectPath("docker-cluster", "test_single_record_table"))
-            .getUnresolvedSchema()
-            .getColumns()
-            .stream()
-            .map(Schema.UnresolvedColumn::getName)
-            .toArray(String[]::new);
+                .getUnresolvedSchema()
+                .getColumns()
+                .stream()
+                .map(Schema.UnresolvedColumn::getName)
+                .toArray(String[]::new);
+
         for (int i = 0; i < actualColumnNames.length - 1; ++i) {
             // Elastic driver returns columns in alphabetical order.
             assertTrue(actualColumnNames[i].compareTo(actualColumnNames[i + 1]) < 0);
         }
     }
+
+    @Test
+    public void testGetTableTimeAttributesProctime() throws TableNotExistException {
+        // given
+        String url = String.format("jdbc:elasticsearch://%s:%d", container.getHost(),
+                container.getElasticPort());
+        Map<String, String> properties = new HashMap<String, String>();
+        properties.put("properties.timeattribute.test_multiple_records_table.proctime.column", "my_proctime");
+
+        // when
+        ElasticCatalog catalog = new ElasticCatalog("test-catalog", "test-database", USERNAME, PASSWORD, url, properties);
+        CatalogBaseTable table = catalog.getTable(new ObjectPath("docker-cluster", "test_multiple_records_table"));
+
+        // then
+        Schema schema = table.getUnresolvedSchema();
+        assertNotNull(table);
+        assertNotNull(schema);
+
+        Schema expectedSchema = Schema.newBuilder().fromFields(
```

Comment:
Please check only if proctime column is defined. I think there is no need to check the entire schema

Reply:
Fixed :)